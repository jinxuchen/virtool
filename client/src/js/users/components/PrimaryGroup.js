import React from "react";
import { map, capitalize } from "lodash-es";
import { connect } from "react-redux";
import { InputError } from "../../base";
import { editUser } from "../actions";

export class PrimaryGroup extends React.Component {
    handleSetPrimaryGroup = e => {
        const value = e.target.value === "none" ? "" : e.target.value;
        this.props.onSetPrimaryGroup(this.props.detail.id, value);
    };

    render() {
        const groupOptions = map(this.props.groups, groupId => (
            <option key={groupId} value={groupId}>
                {capitalize(groupId)}
            </option>
        ));
        return (
            <div>
                <label>Primary Group</label>

                <InputError type="select" value={this.props.primary_group} onChange={this.handleSetPrimaryGroup}>
                    <option key="none" value="none">
                        None
                    </option>
                    {groupOptions}
                </InputError>
            </div>
        );
    }
}
export const mapStateToProps = state => ({
    group: state.users.detail.group,
    primary_group: state.users.detail.primary_group
});

export const mapDispatchToProps = dispatch => ({
    onSetPrimaryGroup: (userId, groupId) => {
        dispatch(editUser(userId, { primary_group: groupId }));
    }
});
export default connect(
    mapStateToProps,
    mapDispatchToProps
)(PrimaryGroup);
