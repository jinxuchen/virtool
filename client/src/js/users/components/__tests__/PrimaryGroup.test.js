import { mapStateToProps, mapDispatchToProps, PrimaryGroup } from "../PrimaryGroup";

describe("<PrimaryGroup />", () => {
    const props = {
        onSetPrimaryGroup: jest.fn(),
        detail: { id: "foo" },
        groups: { foo: "bar" },
        primary_group: "fee"
    };

    it("should render", () => {
        const wrapper = shallow(<PrimaryGroup {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});

describe("mapStateToProps", () => {
    const state = {
        users: {
            detail: { group: "foo", primary_group: "bar" }
        }
    };
    it("should return props", () => {
        const result = mapStateToProps(state);
        expect(result).toEqual({
            group: "foo",
            primary_group: "bar"
        });
    });
});
describe("mapDispatchToProps", () => {
    it("should return editUser in props ", () => {
        const dispatch = jest.fn();
        const props = mapDispatchToProps(dispatch);
        props.onSetPrimaryGroup("foo", "bar");
        expect(dispatch).toHaveBeenCalledWith({
            type: "EDIT_USER_REQUESTED",
            update: {
                primary_group: "bar"
            },
            userId: "foo"
        });
    });
});
