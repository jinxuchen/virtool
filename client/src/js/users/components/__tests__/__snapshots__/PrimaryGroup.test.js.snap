// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrimaryGroup /> should render 1`] = `
<div>
  <label>
    Primary Group
  </label>
  <InputError
    onChange={[Function]}
    type="select"
    value="fee"
  >
    <option
      key="none"
      value="none"
    >
      None
    </option>
    <option
      key="bar"
      value="bar"
    >
      Bar
    </option>
  </InputError>
</div>
`;
